### PR TITLE
fix: Remove escaping of ' in a CtLiteral<String>

### PR DESCRIPTION
**Pull request by MartinWitt**
Currently, spoon escapes every `'` in a `CtLiteral<String>` without a source position. In a `string `escaping `'` is not necessary.

### CODE REVIEW

**Comment by MartinWitt** [comment]
I could create another method overload* to fix the constant value warning. But this would increase the size of the class unnecessary. Any opinions? 

* Actually this is not a method overload because the methods are static

**Comment by I-Al-Istannen** [comment]
> `	 * This exists for backwards compatibility`

The method seems  to be package private, why do you need to keep it if it is never called?

**Comment by MartinWitt** [comment]
I thought about the one guy creating his code inside the same package in his project. I can remove this method if we ignore that one guy.

**Comment by I-Al-Istannen** [comment]
Yes, I would ignore them.

File: src/main/java/spoon/reflect/visitor/LiteralHelper.java [hunk 1]
```diff
@@ -155,7 +159,7 @@ static String getStringLiteral(String value, boolean mayContainsSpecialCharacter
 		} else {
 			StringBuilder sb = new StringBuilder(value.length() * 2);
 			for (int i = 0; i < value.length(); i++) {
-				appendCharLiteral(sb, value.charAt(i), mayContainsSpecialCharacter);
+				appendCharLiteral(sb, value.charAt(i), true, true);
```
Comment (MartinWitt) [inline src/main/java/spoon/reflect/visitor/LiteralHelper.java:162]: This fixes the qodana warning, I have no clue if this is a better code style.

**Review by I-Al-Istannen — CHANGES_REQUESTED** [review]

**Review by I-Al-Istannen — APPROVED** [review]

**Pull request merged**